x-pack/plugins/security_solution/public/common/components/alert_count_by_status/components/multiselect_popover/multiselect_popover.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { EuiFilterButton, EuiFilterSelectItem, EuiPopover } from '@elastic/eui';
+import { euiStyled } from '@kbn/kibana-react-plugin/common';
+
+export interface MultiSelectPopoverProps {
+  title: string;
+  allItems: string[];
+  selectedItems: string[];
+  onSelectedItemsChange: (newItems: string[]) => void;
+}
+
+const StyledFilterButton = euiStyled(EuiFilterButton)`
+  border-radius: 5px;
+  border: 1px solid rgba(17,43,134,0.1); 
+`;
+
+export const MultiSelectPopover = React.memo(
+  ({ allItems, selectedItems, title, onSelectedItemsChange }: MultiSelectPopoverProps) => {
+    const [isItemPopoverOpen, setIsItemPopoverOpen] = useState(false);
+
+    const onChange = useCallback(
+      (item: string) => onSelectedItemsChange(getUpdatedSelectedItems(item, selectedItems)),
+      [selectedItems, onSelectedItemsChange]
+    );
+
+    const itemList = useMemo(() => {
+      return allItems.map((item, index) => (
+        <EuiFilterSelectItem
+          checked={selectedItems.includes(item) ? 'on' : undefined}
+          key={`${index}-${item}`}
+          onClick={() => onChange(item)}
+          title={item}
+        >
+          {item}
+        </EuiFilterSelectItem>
+      ));
+    }, [allItems, selectedItems, onChange]);
+
+    const togglePopover = useCallback(
+      (toState?: boolean) => {
+        setIsItemPopoverOpen((s) => (toState ? toState : !s));
+      },
+      [setIsItemPopoverOpen]
+    );
+
+    return (
+      <EuiPopover
+        ownFocus
+        button={
+          <StyledFilterButton
+            grow={false}
+            data-test-subj={'multiselect-popover-button'}
+            iconType="arrowDown"
+            onClick={() => togglePopover()}
```

Comment:
```suggestion
            onClick={togglePopover}
```

Reply:
I had initially implemented like this but the `FilterButton` component doesnt like this :( It's typing expects a mousehandler

```
onClick?: React.MouseEventHandler<HTMLButtonElement> | React.MouseEventHandler<HTMLAnchorElement> | undefined
```

So I did it as above instead, alternatively I could `ts-ignore`?

Reply:
this is fine, thanks for explaining